test/helpers.ts

```diff
@@ -19,20 +19,14 @@ export interface PromisifiedResponse {
 
 export function httpGet(
   url: string,
-  options?: RequestOptions
+  options: RequestOptions = {}
 ): Promise<PromisifiedResponse> {
   const parsedUrl = new URL(url)
-  const resolvedOptions = Object.assign(
-    {},
-    getRequestOptionsByUrl(parsedUrl),
```

Comment:
This means that we are unable to correctly construct `ClientRequest` if its options is `urlToHttpOptions(url)` on the developer's side:

```js
http.request(urlToHttpOptions(new URL('http://localhost')))
```

Due to the return value of this utility, the options passed to `ClientRequest` will be considered a _legacy URL_, and handled in some way that must be distorting the already complete request options: 

https://github.com/mswjs/interceptors/blob/0c0a4f12d8ad8c3887b33f5e777b84310a34ad5c/src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts#L168-L170

Reply:
Confirming that it's still okay in userland: https://github.com/mswjs/interceptors/pull/618/commits/5dc1e5b366c4e9989e02d32ab5f4885908c0e621.